### PR TITLE
Add 'patch' method to the 'Fork' structure

### PR DESCRIPTION
**Pull request by DarkEld3r**
Funny thing: in the #371 pull request I actually forgot to add getter for `Patch`. 😅

### CODE REVIEW

File: exonum/src/storage/db.rs [hunk 1]
```diff
@@ -431,6 +431,11 @@ impl Fork {
         self.patch
     }
 
+    /// Returns reference to the inner `Patch`.
+    pub fn path(&self) -> &Patch {
```
Comment (dmitry-timofeev) [inline exonum/src/storage/db.rs]: It's supposed to be pat**c**h`, isn't it?

**Comment by DarkEld3r** [comment]
@defuz Probably, I should add wrappers for iterators, right?

**Comment by DarkEld3r** [comment]
One more thing to discuss: we have `Iterator` trait in the storage, so its name is the same as for the standard one (`std::iter::Iterator`). Is that a problem?

On the one hand, it is a little confusing and not very convenient to use something like `use std::iter::Iterator as StdIterator;`.
On the other hand, many crates have their own `Error` or `Result` definitions.

Opinions?

**Comment by ozkriff** [comment]
> it is a little confusing and not very convenient to use something like use `std::iter::Iterator as StdIterator;`.

And it may cause some strange error messages I think.

> On the other hand, many crates have their own `Error` or `Result` definitions.

`std::error::Error` trait is not in the prelude and non-std `Result`s and `Error`s are usually just structures or aliases, not traits.

**Pull request merged**